### PR TITLE
Retrieve schedules from get_schedule_class()

### PR DESCRIPTION
**Pull request by H-Huang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #605
* #604
* __->__ #595

Removes glue code and allow users to pass in schedule arg as a string without updating titan. 

Depends on https://github.com/pytorch/pytorch/pull/137103

### CODE REVIEW

File: torchtitan/config_manager.py [hunk 1]
```diff
@@ -299,14 +299,14 @@ def __init__(self):
         self.parser.add_argument(
             "--experimental.pipeline_parallel_schedule",
             type=str,
-            choices=["1f1b", "gpipe", "interleaved_1f1b", "flexible_interleaved_1f1b"],
+            choices=["1f1b", "gpipe", "Interleaved1F1B", "FlexibleInterleaved1F1B"],
```
Comment (tianyu-l) [inline torchtitan/config_manager.py]: any reason we leave `1f1b` and `gpipe` in lower cases?

Comment (H-Huang) [reply torchtitan/config_manager.py]: The `get_schedule()` method is case insensitive, the main change was i needed to remove the "_" in the other schedules. But I can change these as well for consistency.

**Review by XilunWu — APPROVED** [review]

File: torchtitan/config_manager.py [hunk 2]
```diff
@@ -299,14 +299,14 @@ def __init__(self):
         self.parser.add_argument(
             "--experimental.pipeline_parallel_schedule",
             type=str,
-            choices=["1f1b", "gpipe", "interleaved_1f1b", "flexible_interleaved_1f1b"],
-            default="1f1b",
+            choices=["1F1B", "Gpipe", "Interleaved1F1B", "FlexibleInterleaved1F1B"],
```
Comment (tianyu-l) [inline torchtitan/config_manager.py]: very tiny nit: `GPipe` instead of `Gpipe`

Comment (H-Huang) [reply torchtitan/config_manager.py]: updated

**Pull request merged**